packages/select/src/common/listItemsProps.ts

```diff
@@ -12,12 +12,21 @@ import { ItemListPredicate, ItemPredicate } from "./predicate";
 /** Reusable generic props for a component that operates on a filterable, selectable list of `items`. */
 export interface IListItemsProps<T> extends IProps {
     /**
-     * The currently focused item for keyboard interactions, or `null` to
+     * The currently focused item for keyboard interactions, or `undefined` to
      * indicate that no item is active. If omitted, this prop will be
      * uncontrolled (managed by the component's state). Use `onActiveItemChange`
      * to listen for updates.
      */
-    activeItem?: T | null;
+    activeItem?: T;
+
+    /**
+     * Equality test implementation to determine if two values are representative of the same item.
+     * If not specified, then simple strict equality is used by default.
+     * @param valueA - A value.
+     * @param valueB - Another value.
+     * @return true if the two values are equivalent.
+     */
+    areValuesEqual?: (valueA: T, valueB: T) => boolean;
```

Comment:
⭐️ let's call this `itemsEqual` to match the other `item*` props.

Reply:
would be _super dope_ to support the `| keyof T` syntax like `itemDisabled`, so i could say `itemsEqual="id"` as shorthand for matching `id` props.

Reply:
Done and done. I decided to implement the behavior of processing the `itemsEqual` prop as a utility function in this file (rather than a private method of QueryList) so that its behavior can be easily reused if necessary, and it is more easily/thoroughly unit tested.